pravega-client-examples/src/main/java/io/pravega/example/streamprocessing/EventGenerator.java

```diff
@@ -0,0 +1,92 @@
+
+/**
+ * A simple example app to write messages to a Pravega stream.
+ */
+public class EventGenerator {
+    private static final Logger log = LoggerFactory.getLogger(EventGenerator.class);
+
+    private final AppConfiguration config;
+
+    public static void main(String[] args) throws Exception {
+        final EventGenerator app = new EventGenerator(new AppConfiguration(args));
+        app.run();
+    }
+
+    public EventGenerator(AppConfiguration config) {
+        this.config = config;
+    }
+
+    public AppConfiguration getConfig() {
+        return config;
+    }
+
+    public void run() throws Exception {
+        final ClientConfig clientConfig = ClientConfig.builder().controllerURI(getConfig().getControllerURI()).build();
+        try (StreamManager streamManager = StreamManager.create(getConfig().getControllerURI())) {
+            streamManager.createScope(getConfig().getScope());
+            StreamConfiguration streamConfig = StreamConfiguration.builder()
+                    .scalingPolicy(ScalingPolicy.byEventRate(
+                            getConfig().getTargetRateEventsPerSec(),
+                            getConfig().getScaleFactor(),
+                            getConfig().getMinNumSegments()))
+                    .build();
+            streamManager.createStream(getConfig().getScope(), getConfig().getStream1Name(), streamConfig);
+            streamManager.updateStream(getConfig().getScope(), getConfig().getStream1Name(), streamConfig);
+        }
+
+        final Random rand = new Random(42);
+        @Cleanup final EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(getConfig().getScope(), clientConfig);
+        @Cleanup final EventStreamWriter<SampleEvent> writer = clientFactory.createEventWriter(
+                     getConfig().getStream1Name(),
+                     new JSONSerializer<>(new TypeToken<SampleEvent>(){}.getType()),
+                     EventWriterConfig.builder().build());
+        long sequenceNumber = 0;
+        long sum = 0;
+        final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+        for (;;) {
+            sequenceNumber++;
+            final SampleEvent event = new SampleEvent();
+            event.sequenceNumber = sequenceNumber;
+            event.routingKey = String.format("%3d", rand.nextInt(1000));
+            event.intData = rand.nextInt(1000);
+            sum += event.intData;
+            event.sum = sum;
+            event.timestamp = System.currentTimeMillis();
+            event.timestampStr = dateFormat.format(new Date(event.timestamp));
+            log.info("{}", event);
+            final CompletableFuture<Void> writeFuture = writer.writeEvent(event.routingKey, event);
+            final long ackedSequenceNumber = sequenceNumber;
+            writeFuture.thenRun(() -> log.debug("Acknowledged: sequenceNumber={}", ackedSequenceNumber));
+            Thread.sleep(1000);
+        }
+    }
+
```

Comment:
Do we need to handle `addShutdownHook` here?

Reply:
I don't think we need to handle this here. There is an `addShutdownHook` in `AtLeastOnceApp::run` that will gracefully shutdown (remove itself from the RG) when a user presses control-C or kills the process without `-9`. We don't need to do anything special to terminate the `EventGenerator` app.